NextButtonWidget.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class NextButtonWidget : MonoBehaviour
+{
+    public Text MyLabel;
+    public bool AllowClicks = true;
+    public float ClickTimeout;
+    public GameObject MySceneController;
+    public string ResetURL;
+    private SceneController _MySceneController;
+    private bool FirstUpdate = true;
+    [SerializeField]
+    private Color _OriginalColor;
+    public bool AttemptRestart;
+    public bool SimWantsToRestart;
+    private bool HaveSound;
+    private bool UserIsMashingNext = false;
+
+    // Use this for initialization
+    void Start()
+    {
+        _MySceneController = MySceneController.GetComponent<SceneController>();
+
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (FirstUpdate)
+        {
+            ExposeMyCapi();
+            FirstUpdate = false;
+        }
+
+    }
+
+    public IEnumerator TimeOutUserClicks()
+    {
+        gameObject.GetComponent<Button>().interactable = false;
+        //gameObject.GetComponent<Image>().color = gameObject.GetComponent<Button>().colors.disabledColor;
+
+        yield return new WaitForSeconds(ClickTimeout);
+
+        AllowClicks = true;
+        gameObject.GetComponent<Button>().interactable = true;
+
+        yield return null;
+    }
+
+    public void TriggerNextEvent()
+    {
+        if (AllowClicks)

```

Comment:
capi provides a callback that you can hook into when the check event is complete so you can use that to wait until you allow the student to click next again instead of an arbitrary time.